stl/inc/xutility

```diff
@@ -3649,19 +3647,76 @@ _OutCtgIt _Copy_memmove(_CtgIt _First, _CtgIt _Last, _OutCtgIt _Dest) {
     }
 }
 
+template <class _CtgIt, class _OutCtgIt>
+_OutCtgIt _Copy_memmove_n(_CtgIt _First, const size_t _Count, _OutCtgIt _Dest) {

```

Comment:
The reason this is a separate function is to avoid the multiplication (division by a constant, which compilers translate to multiplication) implicit in `(_LastPtr - _FirstPtr)` used to calculate the return value when `_OutCtgIt` is not a pointer. I think I've found a way to implement `_Copy_memmove_n` in terms of `_Copy_memmove` without losing that advantage.